### PR TITLE
Copter/Blimp: remove stick gesture compass calibration start and stop

### LINKED ISSUE #5062: Add parameter for disabling Stick based Compass calibration
#### Issue details

Stick initiated compass calibration is going to be a new feature of Copter. There should be a parameter associated to allow users and manufacturers to disable it.
If it is enabled with no disabling option there will be a lot of people inadvertently activating it and increasing the APM complexity perception.

Ref Pull
https://github.com/ArduPilot/ardupilot/pull/5061
#### Platform

[X] All
[  ] AntennaTracker
[  ] Copter
[  ] Plane
[  ] Rover

### PR DESCRIPTION
**Pull request by IamPete1**
Fixes #5062

Flight options is not a great place for this, but I don't think its worth adding a new options param just for this.

### CODE REVIEW

**Review by rmackay9 — APPROVED** [review]
I guess we don't want to just remove the stick gestures for compass cal?  I wonder how many users really use it.

**Comment by peterbarker** [comment]
> I guess we don't want to just remove the stick gestures for compass cal? I wonder how many users really use it.

... and most that do use it are using it as a cargo-cult sort of thing, I'd wager.

I'd like to see them disappear.

We could/should add the option to the frsky mavlite handling code to start the compass calibration - it gives the flexibility of doing the cal from the transmitter but with a well-spelt out action rather than a non-discoverable stick gesture.

**Comment by rmackay9** [comment]
@peterbarker an auxiliary switch to trigger the compass calibration seems like a good alternative to me... buttons and lua scripts could also trigger it then.

I vote for us removing the gesture for 4.2.  Sadly we probably won't find out from users if they are unhappy with this choice for some time though.

**Comment by IamPete1** [comment]
I guess step one could be to invert the option, so it would be disabled for everyone and then enabled by the option. Then we would hear how many are using it but they can easily turn it back on. Then if no one complains that it's gone we can remove it completely.

**Comment by rmackay9** [comment]
@IamPete1,

"Out with the old, in with the new" is my opinion on this.  We shouldn't get stuck in the past and be too afraid of removing stuff that we think nobody is using.  useless features add complexity that everyone has to pay the price for... but it's up to you.

**Comment by IamPete1** [comment]
This now moves both the start and stop compass calibration stick gestures from copter and blimp. This simplification means we can directly call the AP_Compass calibration update function from the scheduler, as it is on plane and rover.

**Review by rmackay9 — APPROVED** [review]
I like it but let's discuss on the weekly dev call as well to get other people's input.

**Review by MichelleRos — APPROVED** [review]

File: ArduCopter/Copter.cpp [hunk 1]
```diff
@@ -166,7 +166,7 @@ const AP_Scheduler::Task Copter::scheduler_tasks[] = {
 #if RPM_ENABLED == ENABLED
     SCHED_TASK(rpm_update,            40,    200),
 #endif
-    SCHED_TASK(compass_cal_update,   100,    100),
+    SCHED_TASK_CLASS(Compass, &copter.compass, cal_update, 100, 100),
```
Comment (peterbarker) [inline ArduCopter/Copter.cpp:169]: Can this be moved to AP_Vehicle now?

**Review by peterbarker — APPROVED** [review]

File: ArduCopter/Copter.cpp (hunk 1 above)
Comment (rmackay9) [reply ArduCopter/Copter.cpp:169]: remember that if you move it you need to check whether its priority is changed and whether that affects anything.

**Comment by rmackay9** [comment]
Merged, thanks!

**Pull request merged**